src/Compression/CompressedReadBufferFromFile.cpp

```diff
@@ -102,17 +104,30 @@ size_t CompressedReadBufferFromFile::readBig(char * to, size_t n)
         size_t size_decompressed = 0;
         size_t size_compressed_without_checksum = 0;
 
-        size_t new_size_compressed = readCompressedData(size_decompressed, size_compressed_without_checksum, false);
+        size_t new_size_compressed = readCompressedDataBlockHold(size_decompressed, size_compressed_without_checksum);
+
+        if (new_size_compressed)
+        {
+            decompress_mode = ICompressionCodec::CodecMode::Asynchronous;
```

Comment:
This deserves a comment what is going on here, i.e. how the returned value "new_size_compressed" is to be interpreted.

Reply:
done